packages/webapp/pages/join.tsx

```diff
@@ -0,0 +1,199 @@
+              <b>{referringUser.name}</b> invites you to use daily.dev
+            </p>
+          </FlexRow>
+          <h1 className="mb-4 text-white typo-giga2">
+            The Homepage
+            <br />
+            <b className="text-cabbage-40">Developers Deserve</b>
+          </h1>
+          <p className="mb-10 text-salt-90 typo-title2">
+            Get one personalized feed for all the knowledge you need. Make
+            learning a daily habit or just do something useful while you&apos;re
+            in endless meetings 😜
+          </p>
+          <Button
+            className="max-w-sm h-16 btn-primary !typo-title3"
+            tag="a"
+            href={downloadBrowserExtension}
+            target="_blank"
+            buttonSize={ButtonSize.XLarge}
+          >
+            <FlexCentered className="gap-2">
+              <BrowsersIcon
+                width="50px"
+                height="23px"
+                className="text-theme-label-primary"
+              />
+              Add to browser - it&apos;s free!
+            </FlexCentered>
+          </Button>
+        </FlexCol>
+        <FlexCol className="relative">
+          <Image
+            style={{
+              width: '100%',
+              maxWidth: '1015px',
+              height: '633px',
+              objectFit: 'contain',
+            }}
+            src={cloudinary.referralCampaign.appScreenshot}
+            alt="Image showing the daily.dev app"
+          />
+          <button
+            className="absolute top-1/2 left-1/2 bg-pepper-90 rounded-full transform -translate-x-1/2 -translate-y-1/2"
+            type="button"
+            onClick={() => {
+              setVideoOpen(true);
+            }}
+          >
+            <PlayIcon className="w-28 h-28" size={IconSize.XXXLarge} />
+          </button>
+        </FlexCol>
+      </FlexRow>
+      {isVideoOpen && (
+        <Modal
+          // eslint-disable-next-line @dailydotdev/daily-dev-eslint-rules/no-custom-color
+          className="px-8 bg-[black]"
+          kind={ModalKind.FlexibleCenter}
+          size={ModalSize.XLarge}
+          isOpen={isVideoOpen}
+          onRequestClose={onVideoModalClose}
+        >
+          <CloseButton
+            buttonSize={ButtonSize.Small}
+            className="top-3 right-3 text-white border-white !absolute !btn-secondary"
+            onClick={onVideoModalClose}
+          />
+          <iframe
+            className="w-full border-none aspect-video"
+            // TODO WT-1412-lego-landing-page r.daily.dev link
+            src="https://www.youtube.com/embed/igZCEr3HwCg"
+            title="YouTube video player"
+            allow="encrypted-media;web-share"
+            allowFullScreen
+          />
+        </Modal>
+      )}
+    </div>
+  );
+};
+
+const validateUserId = (value: string) => !!value && value !== '404';
+
+export const getServerSideProps: GetServerSideProps<PageProps> = async ({
```

Comment:
We'll have to check the performance of SSR in production. I like this approach but let's see how it performs

Reply:
If we are setting cookie on client side then we do not need SSR. Regarding this comment https://github.com/dailydotdev/apps/pull/1853#discussion_r1215299987

Reply:
@idoshamun even though we are setting cookie through client side JS I still left this as SSG page. 

Reasons:
- we need dynamic page so we can read query params (static pages do not have `router.query` until hydration)
- it is much cleaner from error handling perspective to check userid and cid params on the server then to do it on client side
- also we can easily fetch referring user on the server too and then pass to page as props
- regarding performance I added `s-max-age=60` header which makes this page cached for 60 seconds on [vercel CDN](https://vercel.com/docs/concepts/edge-network/caching#server) (same as ISR on squad invite page)
  - I did this many times on Vercel with Next.js and can vouch performance is going to be very close or same as static

Reply:
Updated cache to 1 day, since data for referral does not change that oftenand we can profit even more from cached requests.